### PR TITLE
TLV generation fixes

### PR DESCRIPTION
**Pull request by rustyrussell**
More adventures in gossip tlv.

This lets us parse https://github.com/lightningnetwork/lightning-rfc/pull/557#issuecomment-515346187 and generate correct code!

### CODE REVIEW

File: tools/generate-wire.py [hunk 1]
```diff
@@ -122,17 +134,27 @@ def __init__(self):
...
-                self.len_fields[len_field.name] = len_field
-
+                if count in self.fields:
+                    len_field = self.find_data_field(count)
+                    field.add_len_field(len_field)
+                    self.len_fields[len_field.name] = len_field
+                else:
+                    # '...' means "rest of TLV"
+                    assert implicit_len_ok
+                    assert count == '...'
```
Comment (niftynei) [inline tools/generate-wire.py]: can you add an example of this to `tools/test/test_cases`?

File: tools/gen/print_header_template [hunk 2]
```diff
@@ -17,4 +17,10 @@ void print${options.enum_name}_tlv_message(const char *tlv_name, const u8 *msg);
 void printwire_${msg.name}(const char *fieldname, const u8 *cursor);
 
 % endfor
+
+% if options.expose_subtypes:
```
[the remark this answers is not in the document]
Comment (rustyrussell) [reply tools/gen/print_header_template]: No, there are standalone subtypes for gossip queries: they are only in other messages as encoded byte arrays. That's how I found this one!

[the remark this answers is not in the document]
Comment (niftynei) [reply tools/gen/print_header_template]: :astonished:

**Comment by rustyrussell** [comment]
Rebased for minor conflict, test case added.

**Comment by niftynei** [comment]
ACK https://github.com/ElementsProject/lightning/pull/2849/commits/c801efb8955c0457e8504955d3cf2aaae9bdb6f8

**Pull request merged**